frame/utility/src/lib.rs

```diff
@@ -323,6 +330,39 @@ pub mod pallet {
 			let base_weight = T::WeightInfo::batch_all(calls_len as u32);
 			Ok(Some(base_weight + weight).into())
 		}
+
+		/// Dispatches a function call with a provided origin.
+		///
+		/// The dispatch origin for this call must be _Root_.
+		///
+		/// # <weight>
+		/// - O(1).
+		/// - Limited storage reads.
+		/// - One DB write (event).
+		/// - Weight of derivative `call` execution + T::WeightInfo::dispatch_as().
+		/// # </weight>
+		#[pallet::weight({
+			let dispatch_info = call.get_dispatch_info();
+			(
+				T::WeightInfo::dispatch_as()
+					.saturating_add(dispatch_info.weight)
+					// AccountData for inner call origin accountdata.
+					.saturating_add(T::DbWeight::get().reads_writes(1, 1)),
+				dispatch_info.class,
+			)
+		})]
+		pub fn dispatch_as(
+			origin: OriginFor<T>,
+			as_origin: Box<T::PalletsOrigin>,
```

Comment:
Why do we need to box it? It wasn't boxed before you added the benchmark. We should solve it there when possible.

Reply:
it is generally good idea to box big data structure in enum to avoid increase call size. the same reason box is used elsewhere.

Reply:
I see. I thought it was related to the benchmarks. Didn't know that this is a big argument, though.